msteams-platform/tabs/how-to/app-caching-for-your-tab-app.md

```diff
@@ -0,0 +1,219 @@
+
+This process consists of two stages for an app that is enabled for app caching:
+
+1. [First launch of the app](#first-launch-of-the-app)
+1. [Reload app from cache](#reload-app-from-cache)
+
+### First launch of the app
+
+The following flow diagram shows how the app loads for the first time it's launched in the Teams. The app registers the `load` or `beforeUnload`:
+
+:::image type="content" source="../../assets/images/saas-offer/first-launch-app.png" alt-text="This screenshot shows the flow of the first launch of the app in Teams .":::
+
+| # | Interaction | What's going on |
+| --- | --- | --- |
+| 1. | Teams client → App | Teams client launches the app to the Teams. |
+| 2. | App → Teams client | The app initializes the flow. |
+| 3. | App → Teams client | The app registers `Load` or `beforeUnload` handler with Teams clients and caches the app data in the Teams. |
+| 4. | App → Teams client | The app invokes SDK `notifySuccess` to notify Teams client that the initialization flow is complete. |
+| 5. | Teams client → App | When the participants move away from the app, Teams client disposes resources and performs any cleanup needed in the `beforeUnload` handler. |
+| 6. | App → Teams client | The app invokes the `readyToUnload` callback to notify Teams client that the app unload flow is complete. |
+
+### Reload app from cache
+
+The following flow diagram shows how a cached app is reloaded:
+
+:::image type="content" source="../../assets/images/saas-offer/cached-launch-app.png" alt-text="This screenshot shows the flow of the cached launch of the app in Teams.":::
+
+| # | Interaction | What's going on |
+| --- | --- | --- |
+| 1. | Teams client → App | Teams client loads the app and invokes `Load` callback. |
+| 2. | App → Teams client | The app  uses `contentUrl` and `entityId` to route to the desired page, and then invokes SDK `notifySuccess` to notify Teams client that the initialization flow is complete. |
+| 3. | Teams client → App | When the participants move away from the app, Teams client disposes resources and performs any cleanup needed in the `beforeUnload` handler. |
+| 4. | App → Teams client | The app invokes the `readyToUnload` callback to notify Teams client that the app unload flow is complete. |
+
+## Enable app caching
+
+To enable app caching for your app, in the `app-cache-tab.tsx` file (or the equivalent file in your app):
+
+1. Pass `contentUrl` and `entityId` into the load handler to route to the correct page within your app. It also invokes `notifySuccess` or `notifyFailure` to notify Teams client when the app initialization flow is complete.
+   * [contentUrl](create-tab-pages/configuration-page.md#modify-or-remove-a-tab): Add content page URL.
+   * [entityId](create-tab-pages/configuration-page.md#modify-or-remove-a-tab): Add a unique identifier.
+
+1. Call `teamsCore.registerBeforeUnloadHandler` and `teamsCore.registerOnLoadHandler` APIs.
```

Comment:
This should be the first item in the list.

Reply:
Updated as suggested